eml/screens/excercise/ExerciseScreen.jsx

```diff
@@ -36,42 +36,48 @@ export default function ExerciseScreen({ exerciseObject, sectionObject, courseOb
 	const [selectedAnswer, setSelectedAnswer] = useState(null);
 	const [buttonClassName] = useState('');
 	const [showFeedback, setShowFeedback] = useState(false);
-	const [buttonText, setButtonText] = useState('Confirmar Resposta'); 
+	const [buttonText, setButtonText] = useState(null); 
 	const [isPopUpVisible, setIsPopUpVisible] = useState(false); 
 	const [isCorrectAnswer, setIsCorrectAnswer] = useState(false);
 	const [points, setPoints] = useState(10);
 	const [attempts, setAttempts] = useState(0);
 
-	const handleAnswerSelect = (answerIndex) => {
-		setSelectedAnswer(answerIndex);
-		// No need to reset showFeedback or buttonText here since they are handled in handleReviewAnswer
-	};
+	
 
-	async function handleReviewAnswer(isAnswerCorrect) {
-		if (buttonText === 'Confirmar Resposta') {
+	async function handleReviewAnswer(isAnswerCorrect, answerIndex) {
+		setSelectedAnswer(answerIndex);
+		if (buttonText === null) {
+			setButtonText('Continuar');
 			setShowFeedback(true);
 			if (isAnswerCorrect) {
 				setIsCorrectAnswer(true);
-				setButtonText('Continuar');
 				// Award points based on number of attempts
 				setPoints(attempts === 0 ? 10 : 5);
-				// const obj = await completeComponent(exerciseObject, courseObject.courseId, isAnswerCorrect);
 				setIsPopUpVisible(true);
 			} else {
 				setIsCorrectAnswer(false);
-				setButtonText('Tentar Novamente');
 				setAttempts(attempts + 1);
 			}
-		} else if (buttonText === 'Tentar Novamente') {
-			// Reset selectedAnswer and feedback
-			setSelectedAnswer(null);
-			setShowFeedback(false);
-			setButtonText('Confirmar Resposta');
-		} else if (buttonText === 'Continuar') {
+		}
+		if (buttonText === 'Continuar') {
 			setIsPopUpVisible(false);
-			await completeComponent(exerciseObject, courseObject.courseId, true);
-			if(onContinue()){
+      
+			// Check if it is the last component in the section
+			const currentIndex = sectionObject.components.findIndex(component => component.compId === exerciseObject._id);
+			const lastComponent = currentIndex === sectionObject.components.length - 1 ? true : false;
+
+			// If the answer is correct and it is the last component in the section, handleLastComponent is called
+			if (isAnswerCorrect && lastComponent) {
```

Comment:
Group Tree 
PR #325 - 254 Duolingo logic in exercises is broken in dev. 

Is broken with these lines because when a failed question is appended, the amount of exercises increases but the current index is gotten from db or something? @LucaPurcilly

Reply:
Can you explain what exactly is broken?

Reply:
The issue occurs when you answer incorrectly on an earlier question, and answer the orginal last component corrrectly, then the section is completed.

There seems to be an issue with "lastComponent" not being updated, when a user answers an earlier question incorrectly.

@BabaWolo

Reply:
Bug fix - check PR #340 @LucaPurcilly @Andersraae